### ISSUE TITLE
Multiple business-hours

### ISSUE THREAD

**Issue by matteoconsoli** [kylekatarnls/business-time#56]
Any tip on how to implement having multiple business hours ?

**Comment by kylekatarnls** [comment]
Hello,

It's possible to use multiple instance of `Spatie\OpeningHours\OpeningHours`:

```php
use Spatie\OpeningHours\OpeningHours;

$hours1 = OpeningHours::create([
    'monday'     => ['09:00-12:00', '13:00-18:00'],
]);
$hours2 = OpeningHours::create([
    'monday'     => ['08:00-11:00', '12:00-17:30'],
]);

$date = CarbonImmutable::parse('2022-12-12 08:30');

var_dump($hours1->isOpenAt($date));
var_dump($hours2->isOpenAt($date));
```

[Demo](https://play.phpsandbox.io/nesbot/carbon?version=2.64.0.0&input=%3C%3Fphp%0A%0Ause%20Spatie%5COpeningHours%5COpeningHours%3B%0A%0A%24hours1%20%3D%20OpeningHours%3A%3Acreate(%5B%0A%20%20%20%20%27monday%27%20%20%20%20%20%3D%3E%20%5B%2709%3A00-12%3A00%27%2C%20%2713%3A00-18%3A00%27%5D%2C%0A%5D)%3B%0A%24hours2%20%3D%20OpeningHours%3A%3Acreate(%5B%0A%20%20%20%20%27monday%27%20%20%20%20%20%3D%3E%20%5B%2708%3A00-11%3A00%27%2C%20%2712%3A00-17%3A30%27%5D%2C%0A%5D)%3B%0A%0A%24date%20%3D%20CarbonImmutable%3A%3Aparse(%272022-12-12%2008%3A30%27)%3B%0A%0Avar_dump(%24hours1-%3EisOpenAt(%24date))%3B%0Avar_dump(%24hours2-%3EisOpenAt(%24date))%3B%0A)

Most mixin methods match a method with the same name in `OpeningHours`. Custom holidays added as `exceptions` would also work. Sadly for the holidays.region (`'us-ny'`, `de-national`, etc.) it would be a bit more tricky. I don't think it's possible out of the box, but I might extract the calculator from the mixin so it can be run as an instance instead of statically.

**Comment by kylekatarnls** [comment]
Or you can also have local opening hours scoped to a specific Carbon object:
```php
<?php

use Cmixin\BusinessTime;

BusinessTime::enable([Carbon::class, CarbonImmutable::class]);

$date1 = CarbonImmutable::parse('2022-12-12 08:30');
$date1->setOpeningHours([
    'monday'     => ['09:00-12:00', '13:00-18:00'],
]);

$date2 = CarbonImmutable::parse('2022-12-12 08:30');
$date2->setOpeningHours([
    'monday'     => ['08:00-11:00', '12:00-17:30'],
]);

var_dump($date1->isOpen());
var_dump($date2->isOpen());
```
[Demo](https://play.phpsandbox.io/embed/nesbot/carbon?version=2.64.0.0&input=%3C%3Fphp%0A%0Ause%20Cmixin%5CBusinessTime%3B%0A%0ABusinessTime%3A%3Aenable(%5BCarbon%3A%3Aclass%2C%20CarbonImmutable%3A%3Aclass%5D)%3B%0A%0A%24date1%20%3D%20CarbonImmutable%3A%3Aparse(%272022-12-12%2008%3A30%27)%3B%0A%24date1-%3EsetOpeningHours(%5B%0A%20%20%20%20%27monday%27%20%20%20%20%20%3D%3E%20%5B%2709%3A00-12%3A00%27%2C%20%2713%3A00-18%3A00%27%5D%2C%0A%5D)%3B%0A%0A%24date2%20%3D%20CarbonImmutable%3A%3Aparse(%272022-12-12%2008%3A30%27)%3B%0A%24date2-%3EsetOpeningHours(%5B%0A%20%20%20%20%27monday%27%20%20%20%20%20%3D%3E%20%5B%2708%3A00-11%3A00%27%2C%20%2712%3A00-17%3A30%27%5D%2C%0A%5D)%3B%0A%0Avar_dump(%24date1-%3EisOpen())%3B%0Avar_dump(%24date2-%3EisOpen())%3B)

**Comment by kylekatarnls** [comment]
Glad to announce this is now implemented with a dedicated class in [1.12.0](https://github.com/kylekatarnls/business-time/releases/tag/1.12.0)

See [`Schedule` class](https://github.com/kylekatarnls/business-time#multiple-business-hours)
